### PR TITLE
feature/SC-4994-admin-creates-a-class-without-teacher-and-students

### PR DESCRIPTION
**Pull request by Ojself**
Ticket:
https://ticketsystem.schul-cloud.org/browse/SC-4994

depends on this client pr:
https://github.com/schul-cloud/schulcloud-client/pull/1956

# Description

## Approval for review
- [ ] All points were discussed with the ticket creator, support-team or product owner. The code upholds all quality guidelines from the PR-template.

> Notice: Please remove the WIP label if the PR is ready to review, otherwise nobody will review it.

### Link to Definiton of Done
More and detailed information on the *definition of done* can be found [on Confluence](https://docs.schul-cloud.org/pages/viewpage.action?pageId=92831762)

### CODE REVIEW

File: page-objects/administration.js [hunk 1]
```diff
@@ -9,69 +9,109 @@ let name;
...
+        await classCreationExtraOptions.click()
+
+        const classNameInputField = await driver.$(Admin.classNameInputField)
+        await classNameInputField.setValue(className);
+
+        const confirmButton = await driver.$(Admin.confirmClassCreate)
+        await confirmButton.click()
+
+    },
+    verifyNewEmptyClassCreated: async function (className = '11c', numOfStudents = '0') {
```
[the remark this answers is not in the document]
Comment (Parnswir) [reply page-objects/administration.js:45]: these are default parameters; if you call `verifyNewEmptyClassCreated('foobar', undefined)` or leave out one or both parameters, the defaults from the function declaration are used.

It's used in `Then(/^admin should see the class (.*) with (.*) participants.$/`

File: shared-objects/administrationData.js [hunk 2]
```diff
@@ -2,18 +2,26 @@ const { CLIENT } = require("./servers");
...
+  setEmail: 'input[data-testid=\'create_student_input_email\']',
+  sendALinkBox: 'input[data-testid=\'create_student_input_send_link\']',
   namesContainer: 'tbody[data-testid=\'students_names_container\']',
   consentSubmitBtn: 'button[data-testid=\'submit_consent\']',
   birthday_field: 'input[data-testid=\'create_birthday\']',
+<<<<<<< HEAD
+  classCreationExtraOptions: 'a[data-testid=\'classCreationExtraOptions\']'
+=======
   administrationsTabs: 'ul.subitems span'
+>>>>>>> 238dfdf224eab7fc660e3860e5c4bbf238554045
```
Comment (Parnswir) [inline shared-objects/administrationData.js:26]: Unresolved merge conflict

File: shared-objects/loginData.js [hunk 3]
```diff
@@ -10,8 +10,13 @@ module.exports = {
   eligiblePupilPassword: 'schulcloud',
   notEligiblePupilUsername: 'paula.meyer@schul-cloud.org',
   notEligiblePupilPassword: 'Schulcloud1!',
+<<<<<<< HEAD
+  defaultAdminUsername: "admin@schul-cloud.org",
+  defaultAdminPassword: "Schulcloud1!",
+=======
   defaultAdminUsername: 'admin@schul-cloud.org',
   defaultAdminPassword: 'Schulcloud1!',
+>>>>>>> 238dfdf224eab7fc660e3860e5c4bbf238554045
```
Comment (Parnswir) [inline shared-objects/loginData.js:19]: unresolved merge conflict

File: step_definitions/addPupilsToTheCourse-steps.js [hunk 4]
```diff
@@ -6,20 +6,20 @@ const teacherLogin = require('../page-objects/teacherLogin');
...
-Given(/^teacher is logged in successfully$/, function() {
-	return teacherLogin.performLogin(Login.defaultTeacherUsername,Login.defaultTeacherpassword);
+Given(/^teacher is logged in successfully$/, function () {
+	return teacherLogin.performLogin(Login.defaultTeacherUsername, Login.defaultTeacherpassword);
 });
 
-Given('teacher goes to courses page', function() {
+Given('teacher goes to courses page', function () {
 	return createCourse.goToCourses();
 });
```
Comment (Parnswir) [inline step_definitions/addPupilsToTheCourse-steps.js:18]: Why do we have to redefine these steps in every test case? Probably not a problem with your PR, but the repo in general

Comment (Ojself) [reply step_definitions/addPupilsToTheCourse-steps.js:18]: I'm not sure..

Comment (yuliashikhareva) [reply step_definitions/addPupilsToTheCourse-steps.js:18]: Technically it doesn’t matter how you name your step definition files, or which step definitions you put in a file. You could have one giant file containing all your step definitions. However, as the project grows, the file can become messy and hard to maintain. Instead, it is recommended creating a separate *_steps.js file for each domain concept(=feature)

Comment (Parnswir) [reply step_definitions/addPupilsToTheCourse-steps.js:18]: What I mean is, can we import steps that are used in many test cases?

File: CHANGELOG.md [hunk 5]
```diff
@@ -10,6 +10,12 @@ Allowed Types of change: `Added`, `Changed`, `Deprecated`, `Removed`, `Fixed`, `
 
 ### Added
 
+## [23.2.0] - 2020-06-29
+
+### Added in 23.2.0
+
+- Admin can create empty classes
```
Comment (Parnswir) [inline CHANGELOG.md:17]: Please have a look at the specification of the changelog. You should only add your changes to the `[Unreleased]` section (which is later renamed as part of the release process).

File: step_definitions/addPupilsToTheCourse-steps.js (hunk 4 above)
Comment (Ojself) [reply step_definitions/addPupilsToTheCourse-steps.js:18]: This branch has been closed due to name change. See new pr
https://github.com/schul-cloud/integration-tests/pull/88

**Comment by Ojself** [comment]
This branch has been closed due to name change. See new pr
https://github.com/schul-cloud/integration-tests/pull/88

**Pull request closed without merging**